src/loki/search.cc

```diff
@@ -317,8 +332,10 @@ struct bin_handler_t {
 
         // do we want this edge
         if (edge_filter(edge) != 0.0f) {
+          auto edge_heading = get_edge_heading(candidate.edge, *candidate.edge_info, candidate.point,
```

Comment:
instaed of this separate function, can you pass `path_edge` to `heading_filter` and let the heading filter always set the heading. would be less duplicated code and minimize the calls to the tangent_angle function.

Reply:
Good Point!  
But in line 52 we saw ` if (!location.heading_) {
    return false;
  }`  means when passing request we always need to pass headings to return true. Is there an elegant way to deal with it?